src/q8/sigproc-ELEC2900/exam/2019/Juin/All/sigproc-ELEC2900-exam-2019-Juin-All.tex

```diff
@@ -0,0 +1,107 @@
+\end{enumerate}
+
+\nosolution
+
+\section{Questions of Laurent Jacques}
+
+\subsection{LJ1}
+
+We are given the following difference equation that implements a digital filter which is designed starting from a prototype analog filter.
+$$(2-\sqrt{2})y[n] = x[n] - 2 x[n-1] + x[n-2] + (2+\sqrt{2})y[n-2]$$
+\begin{enumerate}
+    \item Recover the analog prototype of this filter.
+    \item The low-pass filter used to produce the analog high-pass prototype is a butterworth filter of order 2, whose cutoff frequency is $\Omega_p = \frac{1}{2}$ has the following expression:
+    $$H(s)=\frac{1}{(s-\e^{\frac{j3\pi}{4}})(s-\e^{\frac{j5\pi}{4}})}$$
+    Give the expression of the corresponding high-pass filter\footnote{The spectral transformation of a filter from low-pass to high-pass is given by $F(\hat{s}) = \frac{\Omega_p\hat{\Omega}_p}{\hat{s}}$.}. Give the corresponding value of $\hat{\Omega}_p$ to implement the filter above.
+    \item What is the cutoff frequency of the digital filter?
+\end{enumerate}
+
+\subsection{LJ2}
+
+\begin{enumerate}
+    \item Remind what are the conditions for a system function $H(z)=\dfrac{\sum_{k=0}^Ma_kz^{-k}}{\sum_{k=0}^Nb_kz^{-k}}$ to be minimum phase.
+    
+    \item Prove mathematically that any rational system function can be written as the product of a minimum phase filter and an all-pass filter.
+    
+    \item As a reminder, the phase lag of a filter is defined as:
+    \begin{equation*}
+        P_L[H](\omega) = - \angle\{H(\e^{j\Omega}) \}
+    \end{equation*}
+    where $\angle \{\e^{j\Omega}\}=\Omega$.
+    
+    Knowing that an all-pass filter has a phase lag which is always positive on the range $[0,\pi]$, prove that among the filters which can produce $|H(\e^{j\Omega})|$, the minimum phase filter is the one with the lowest phase lag on the range $[0,\pi]$.
+\end{enumerate}
+
+\subsection{LJ3}
+
+We are interested in studying the features of the Haar wavelet transform.
+
+\begin{enumerate}
+    \item As a reminder, the \emph{scaling} function is of amplitude 1 between 0 and 1, and the "mother" \emph{wavelet} function is of amplitude 1 between 0 and $\frac{1}{2}$ and -1 between $\frac{1}{2}$ and 1. Deduce what are the impulse responses of the filters $h[n]$ and $g[n]$ used to apply the fast wavelet transform.
+
+    \item Give the schematic which serves to compute the approximation and detail coefficients of the Fast Wavelet Transform for a 1-dimensional signal of $N=2^J$ coefficients. Suppose we can assume the \emph{wavelet crime} and explain this concept.
+    
+    \item What is the computational complexity of this transform? Explain briefly.
+    
+    \item Suppose we have a signal $x[n] \in \mathbb{R}^N$ such that  $x[n] = (x[0],x[1],\dots,x[N-1])$, we build the coefficients $(c[0],c[1],\dots,c[N-1])$, where $c[0]$ is the approximation coefficient at level 0, $c[1]$ is the detail coefficient at level 0, and $(c[2j],\dots,c[2j+1])$ are the detail coefficients at level $j \in (0,J-1)$.
```

Comment:
```suggestion
    \item Suppose we have a signal $x[n] \in \Rn$ such that  $x[n] = (x[0],x[1],\dots,x[N-1])$, we build the coefficients $(c[0],c[1],\dots,c[N-1])$, where $c[0]$ is the approximation coefficient at level 0, $c[1]$ is the detail coefficient at level 0, and $(c[2j],\dots,c[2j+1])$ are the detail coefficients at level $j \in (0,J-1)$.
```

Additionally, is there a particular why the last interval is written with the American notation for open intervals? It seems to me like this could (and should) be a closed interval, but I could be wrong.

Reply:
Don't know but I guess it is correct.